File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.java
import java.io.*;
import java.util.*;

class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        int[] answer = new int [2];
        Arrays.sort(lottos);
        Arrays.sort(win_nums);
        int zero_cnt = 0;
        int correct = 0;
        
        for(int i=0; i < 6; i++){
            if(lottos[i] == 0){
                zero_cnt++;
                continue;
            }
            for(int j=0; j < 6; j++){
                if(lottos[i] == win_nums[j]){
                    correct++;
                    break;
                }
            }
        }
        
        correct += zero_cnt;
        
        for(int i=0; i<2; i++){
            if(correct == 6){
                answer[i] = 1;
            }else if(correct == 5){
                answer[i] = 2;
            }else if(correct == 4){
                answer[i] = 3;
            }else if(correct == 3){
                answer[i] = 4;
            }else if(correct == 2){
                answer[i] = 5;
            }else{
                answer[i] = 6;
            }              
            correct -= zero_cnt;                         
        }     
            
        
        return answer;
    }
}

File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.kt
class Solution {
    fun solution(lottos: IntArray, win_nums: IntArray): IntArray {
        val ansewer = IntArray(2)
        
        var cnt = 0
        var zeroCnt = 0
        
        lottos.filter{it != 0}.toHashSet().also { lottoSet ->
           win_nums.forEach {
               if(lottoSet.contains(it)){
                   cnt++
               }
           }
           zeroCnt = lottos.size - lottoSet.size
        }
        
        ansewer[0] = chkWinner(cnt + zeroCnt)
        ansewer[1] = chkWinner(cnt)
        
        return ansewer
    }
    
    fun chkWinner(cnt : Int) : Int{
        return when(cnt){
            6 -> 1
            5 -> 2
            4 -> 3
            3 -> 4
            2 -> 5
            else -> 6
        }
    }
}